spring-batch-infrastructure/src/test/java/org/springframework/batch/item/kafka/KafkaItemWriterTests.java

```diff
@@ -103,6 +103,19 @@ public void testBasicDelete() throws Exception {
 		verify(this.kafkaTemplate).sendDefault(items.get(1), null);
 	}
 
+	@Test
+	public void testKafkaTemplateCanBeReferencedFromSubclass() {
+		KafkaItemWriter<String, String> kafkaItemWriter = new KafkaItemWriter<String, String>() {
+			@Override
+			protected void writeKeyValue(String key, String value) {
+				this.kafkaTemplate.flush();
+			}
+		};
+		kafkaItemWriter.setKafkaTemplate(kafkaTemplate);
+		kafkaItemWriter.writeKeyValue("k", "v");
+		verify(kafkaTemplate).flush();
```

Comment:
Related to the previous comment, this should be updated to `verify(this.kafkaTemplate).sendDefault("k", "v");`.

Reply:
makes sense. I'll update the PR